src/EprPrnIntegration.Api.UnitTests/UpdateProducersFunctionTests.cs

```diff
@@ -151,8 +183,54 @@ public async Task Run_FailedToUpdateProducers_LogsRawResponseBody()
         _loggerMock.Verify(logger => logger.Log(
             LogLevel.Error,
             It.IsAny<EventId>(),
-            It.Is<It.IsAnyType>((v, t) => v.ToString().Contains($"Failed to parse error response body. Raw Response Body: {responseBody}")),
+            It.Is<It.IsAnyType>((v, t) =>
+                v.ToString().Contains($"Failed to parse error response body. Raw Response Body: {responseBody}")),
             null,
             (Func<It.IsAnyType, Exception, string>)It.IsAny<object>()), Times.Once);
     }
+
+    [Fact]
+    public async Task Run_SendsDeltaSyncExecutionToQueue()
+    {
+        // Arrange
+        var startHour = 18;
+        _configurationMock.Setup(c => c["UpdateProducersStartHour"]).Returns(startHour.ToString());
+        var updatedProducers = new List<UpdatedProducersResponseModel> { new UpdatedProducersResponseModel() };
+
+        _organisationServiceMock
+            .Setup(service => service.GetUpdatedProducers(It.IsAny<DateTime>(), It.IsAny<DateTime>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(updatedProducers);
+
+        _npwdClientMock
+            .Setup(client => client.Patch(It.IsAny<ProducerDelta>(), NpwdApiPath.UpdateProducers))
+            .ReturnsAsync(new HttpResponseMessage { StatusCode = System.Net.HttpStatusCode.OK });
+
+        // Mock DeltaSyncExecution
+        var deltaSyncExecution = new DeltaSyncExecution
+        {
+            SyncType = NpwdDeltaSyncType.UpdatedProducers,
+            LastSyncDateTime = DateTime.UtcNow.AddHours(-1)
+        };
+
+        _serviceBusProviderMock
+            .Setup(provider => provider.ReceiveDeltaSyncExecutionFromQueue(NpwdDeltaSyncType.UpdatedProducers))
+            .ReturnsAsync(deltaSyncExecution);
+
+        _serviceBusProviderMock
+            .Setup(provider => provider.SendDeltaSyncExecutionToQueue(It.IsAny<DeltaSyncExecution>()))
+            .Returns(Task.CompletedTask); // Mock the Send method to complete successfully.
+
+        var function = new UpdateProducersFunction(
+            _organisationServiceMock.Object,
+            _npwdClientMock.Object,
+            _loggerMock.Object,
+            _configurationMock.Object,
+            _serviceBusProviderMock.Object);
+
+        // Act
+        await function.Run(null);
+
+        // Assert
+        _serviceBusProviderMock.Verify(provider => provider.SendDeltaSyncExecutionToQueue(It.Is<DeltaSyncExecution>(d => d.SyncType == NpwdDeltaSyncType.UpdatedProducers)), Times.Once);
+    }
 }
```

Comment:
Can we add test to validate if there is no message on queue it is picking default from config

Reply:
done